crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs

```diff
@@ -729,9 +739,85 @@ where
                         }
                     });
 
-                if !zk_proofs.is_empty() {
-                    warn!("ZK proofs are not empty");
-                    // TODO: Implement this
+                for proof in zk_proofs {
+                    let state_transition = match proof.clone() {
+                        Proof::Full(proof) => Vm::verify_and_extract_output::<
+                            <Da as DaService>::Spec,
+                            Stf::StateRoot,
+                        >(
+                            &proof, &self.code_commitment
```

Comment:
after verification compare sequencer public key and sequencer da public key the node has

Reply:
if they dont match, ignore proof

Reply:
also compare post and pre state roots

if they don't match, put out a warning 

"Prover is on different state"

Reply:
we'll reorg on such occurences